src/common/tests_dmg_helpers.c

```diff
@@ -488,20 +489,25 @@ dmg_pool_create(const char *dmg_config_file,
 					    entry->dpe_str);
 			if (args == NULL)
 				D_GOTO(out, rc = -DER_NOMEM);
-		} else {
-			static bool need_init_rand = true;
+			has_label = true;
+		}
+	}
 
-			if (need_init_rand) {
-				srand((unsigned int)time(NULL));
-				need_init_rand = false;
-			}
+	if (!has_label) {
+		char	label[] = "/tmp/test_XXXXXX";
+		int	tmp_fd;
 
-			/* pool label is required, generate one randomly */
-			args = cmd_push_arg(args, &argcount, "--label=test_%x ",
-					    rand());
-			if (args == NULL)
-				D_GOTO(out, rc = -DER_NOMEM);
+		/* pool label is required, generate a unique one randomly */
+		tmp_fd = mkstemp(label);
```

Comment:
Hmm. I was wondering how this worked... So I guess label is used as a template and the trailing Xs are replaced with random characters? That's clever. The only concern I have with this is that I think tmp_fd needs to be closed and the temp file cleaned up, right?

Reply:
Oops, you're right. I was thinking of using tmpnam instead (the result is similar), but I thought it was nice to leave the file there for the duration of the run so the name couldn't possibly end up duplicated between processes. But that makes the cleanup process harder. We'd need to save the state somewhere.

Reply:
Alternately we could just retry with different labels if it says the label is no good. I just want to limit the ability of daos_test to fail due to duplicate labels.

Reply:
Played around a little with different ideas, but it seemed like a bit too much effort in the long run. So I'm just cleaning up the tmpfile and considering this good enough.